src/address.rs

```diff
@@ -1,174 +1,132 @@
-            }
-        }
-        if (char == '/') & in_character_range {
-            return Err(OscError::BadAddress("Character range (square brackets) was started but not closed before the next address part started"));
-        }
-        if char == ']' {
-            in_character_range = false;
-        }
-        if char == '{' {
-            in_string_list = true;
-        }
-        if (char == '/') & in_string_list {
-            return Err(OscError::BadAddress("String list (curly brackets) was started but not closed before the next address part started"));
-        }
-        if char == '}' {
-            in_string_list = false;
-            if in_character_range {
-                return Err(OscError::BadAddress("Can not start a string list (curly brackets) within character range (square brackets)"));
-            }
-        }
-    }
-    if in_character_range {
-        return Err(OscError::BadAddress("Character range (square brackets) was started but not closed before the end of the address"));
-    }
-    if in_string_list {
-        return Err(OscError::BadAddress(
-            "String list (curly brackets) was started but not closed before the end of the address",
-        ));
-    }
-
-    Ok(())
+fn map_alternative(s: &str) -> String {
+    wrap_with(&s.replace(',', "|"), "(", ")")
 }
 
-/// Match a single part of two OSC addresses
-fn match_part(message_part: &str, method_part: &str) -> bool {
-    // direct match
-    if message_part == method_part {
-        return true;
-    }
-
-    // Use regex for everything else
-    let mut pattern = message_part
-        .to_string()
-        .replace("*", "\\w*")
-        .replace("?", "\\w")
-        .replace("[!", "[^")
-        .replace(",", ")|(?:")
-        .replace("{", "(?:")
-        .replace("}", ")");
-
-    // Anchor the pattern to beginning and end of string
-    pattern = format!("^{}$", pattern);
-
-    let re = Regex::new(pattern.as_str()).unwrap();
+fn wrap_with(s: &str, pre: &str, post: &str) -> String {
+    pre.to_string() + s + post.into()
+}
 
-    re.is_match(method_part)
+fn map_wildcard(_: &str) -> String {
+    r"\w*".into()
 }
 
-/// Check if a message address matches a method address
-pub fn match_address(message_addr: &str, method_addr: &str) -> Result<bool> {
-    match validate_message_address(message_addr) {
-        Ok(()) => {}
-        Err(e) => return Err(e),
-    }
-    match validate_method_address(method_addr) {
-        Ok(()) => {}
-        Err(e) => return Err(e),
-    }
+fn map_question_mark(_: &str) -> String {
+    r"\w?".into()
+}
 
-    let message_addr_parts: Vec<&str> = message_addr.split('/').collect();
-    let method_addr_parts: Vec<&str> = method_addr.split('/').collect();
+fn parse_address_part(input: &str) -> IResult<&str, &str> {
+    // TODO: the second parser, is_not, should require a non empty match
+    preceded(char('/'), is_not(" \t\r\n#*,/?[]{}"))(input)
```

Comment:
```suggestion
    preceded(char('/'), take_till1(|c| " \t\r\n#*,/?[]{}".contains(c)))(input)
```
This will take characters until the condition is met, but at least 1